Siirrä budjetoidut kustannukset toteutuneet työt tauluun

rckopteri opened the pull request:
Menee niin isoksi tuo kustannusten seuranta feature, että yritän pilkkoa sitä osiin.
Tässä kolmas osa. eli budjetoitujen töiden (kustannusarvoitu_tyo ja johto_ja_hallintakorvaus taulujen data) siirto kuukauden viimeisenä päivänä toteutuneeksi kustannukseksi.

Tehtävä on ajastettu ajettavaksi aina öisin. 
Myös vanhat testidatat siirretään.
Uusi data siirretään kuukausittain.

rckopteri committed:
Ajastettu tehtävä: Siirrä budjetoidut kustannukset toteutuneet työt tauluun

rckopteri committed:
Toteutunut_työ tauluun siirretäänkin vain osa kustannusarvoitu_tyo taulun sisällöstä.

Hakuja siis rajattu pariin tehtävään ja yhteen tehtäväryhmään.

rckopteri committed:
Rajaa siirrettävät toteutuneet työt kahteen tehtävään ja tehtäväryhmään myös flyway migraatiossa

rckopteri committed:
Ja vielä kerran vaihdetaan tehtävien id:t koska ne eivät olekaan samat jokaisessa ympäristössä

jarnovayrynen committed:
Merge branch 'develop' into VHAR-3387-toteutuneet-tyot

jarnovayrynen committed:
Rename V1_850__.sql to V1_851__.sql

jarnovayrynen commented on tietokanta/src/main/resources/db/migration/V1_850__.sql:
```diff
@@ -0,0 +1,87 @@
+
+
+COMMENT ON table toteutunut_tyo IS
+    E'Kustannusarvioitu_tyo taulusta siirretään kuukauden viimeisenä päivänä mikäli tehtäväryhmä: Erillishankinnat (W) tai tehtävät: 3054 ja 3055
```
lokaalikannan id:t pois

rckopteri replied:
Poistettu

jarnovayrynen commented on tietokanta/src/main/resources/db/migration/V1_850__.sql:
```diff
@@ -0,0 +1,87 @@
+FROM kustannusarvioitu_tyo k
+WHERE (SELECT (date_trunc('MONTH', format('%s-%s-%s', k.vuosi, k.kuukausi, 1)::DATE))) <
+      date_trunc('month', current_date)
+  AND (k.tehtavaryhma = (SELECT id FROM tehtavaryhma where nimi = 'Erillishankinta (W)')
```
tähän yksilöity tunniste?

rckopteri replied:
Vaihdettu yksilöivään_tunnisteeseen

jarnovayrynen commented on tietokanta/src/main/resources/db/migration/V1_850__.sql:
```diff
@@ -0,0 +1,87 @@
+WHERE (SELECT (date_trunc('MONTH', format('%s-%s-%s', k.vuosi, k.kuukausi, 1)::DATE))) <
+      date_trunc('month', current_date)
+  AND (k.tehtavaryhma = (SELECT id FROM tehtavaryhma where nimi = 'Erillishankinta (W)')
+        OR k.tehtava = (select id from toimenpidekoodi t where t.nimi = 'Toimistotarvike- ja ICT-kulut, tiedotus, opastus, kokousten järjestäminen jne.') -- 3055,"Toimistotarvike- ja ICT-kulut, tiedotus, opastus, kokousten järjestäminen jne."
```
Tähän subqueryyn t.nimi IN ('nimi1', 'nimi2')?

rckopteri replied:
Vaihettu IN () malliseen kyselyyn

jarnovayrynen commented on tietokanta/src/main/resources/db/migration/V1_850__.sql:
```diff
@@ -0,0 +1,87 @@
+);
+
+create index toteutunut_tyo_urakka_index
+    on toteutunut_tyo (urakka_id, kuukausi, vuosi);
```
urakka_id, vuosi, kuukausi

rckopteri replied:
Indeksin järjestys muutettu

jarnovayrynen commented on tietokanta/testidata.sql:
```diff
@@ -176,4 +176,56 @@ SELECT paivita_kaikki_sopimuksen_kaytetty_materiaali();
 -- Paikkaukset
 \i testidata/paikkaukset.sql
 
+-- Siirretään kaikki rivit mikäli tehtäväryhmä: Erillishankinnat (W) tai tehtävät: 3054 ja 3055
```
Tämä siirretään omaan filuun testidata/toteutuneet_tyot/kustannukset.sql-
Ja manifestifiluun viittaus siihen.

rckopteri replied:
Siirretty omaan tiedostoon.

jarnovayrynen commented on tietokanta/src/main/resources/db/migration/V1_850__.sql:
```diff
@@ -0,0 +1,87 @@
+-- toteutunut_tyo tauluun siirretään kustannuksia, yksinkertaistamaan toteutuneiden kustannusten seurantaa
+
+CREATE TABLE toteutunut_tyo
```
Taulun nimeksi toteutuneet_kustannukset (I know..!)

rckopteri replied:
:) Muutettu

jarnovayrynen commented on tietokanta/src/main/resources/db/migration/V1_850__.sql:
```diff
@@ -0,0 +1,87 @@
+    unique (toimenpideinstanssi, tehtava, sopimus_id, urakka_id, vuosi, kuukausi)
+);
+
+create index toteutunut_tyo_urakka_index
```
järjestys voisi olla urakka_id, vuosi, kuukausi

rckopteri committed:
Vaihdetaan toteutunut_tyo toteutuneet_kustannukset tauluksi

Muutetaan toteutuneiden kustannusten automaattinen ajo alkamaan klo 1.40 yöllä. Ja refaktoroidaan vähä koodeja

rckopteri committed:
Vaihdetaan vielä automaattinen siirto tapahtumaan kuukauden ensimmäinen päivä,

koska yöllinen automatic ajo on aina seuraavan päivän puolella. Nyt kaikki budjetoidut kustannukset tulee varmasti mukaan.

rckopteri committed:
Kirjoitusvirheen korjaus

rckopteri committed:
Merge branch 'develop' into VHAR-3387-toteutuneet-tyot

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_851__.sql

rckopteri committed:
Merge branch 'develop' into VHAR-3387-toteutuneet-tyot

rckopteri committed:
Merge confliktin korjaus

rckopteri committed:
Lisää ajastetusta tehtävästä kahva testeihin

rckopteri committed:
Merge branch 'develop' into VHAR-3387-toteutuneet-tyot

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_856__.sql

rckopteri committed:
Ja typon korjaus

rckopteri committed:
Merge branch 'develop' into VHAR-3387-toteutuneet-tyot
